plugins/modules/dconf.py

```diff
@@ -258,6 +258,48 @@ def __init__(self, module, check_mode=False):
         # Check if dconf binary exists
         self.dconf_bin = self.module.get_bin_path('dconf', required=True)
 
+    # GVariant strings can use either single or double quotes, and therefore
+    # the user can use either single or double quotes when specifying a string
+    # value in their task, and what they choose to use may not match what is
+    # output by the dconf command, so for accurate comparison we need to strip
+    # the quotes and just compare what's inside them.
+    #
+    # This doesn't give us 100% accurate GVariant string interpretation, but
+    # it solves many of the common use cases and makes us much more accurate
+    # than if we just compare the two strings directly.
+    @staticmethod
+    def gvariant_maybe_string(value):
```

Comment:
I think it would be best to add unit tests for this. That will also increase trust that this code really does what it should.

(TBH I'm kind of surprised that this module really expects users to provide data in the exact form that the program requires them. I would really expect that the module does know how to parse / format data the way dconf wants it. That would make the module a lot more useful and user-friendly in my opinion.)